### PR TITLE
feat(ci): enable build Debian packages of external Magma dependencies  

### LINKED ISSUE #13466: Build debian artifacts for Magmas external dependencies
## Story

We want to build external dependencies ourselves and provide debian packages so that we do not rely on manually build and uploaded artifacts. The actual CI setup will be done in #14296.

Probably not all artifacts can currently be built by `third_party/build/build.py` - this needs to be checked and fixed in scope of this issue. (at least `aioh2-build.sh` is not even executable and the suffix needs to be `_build`).

## Acceptance Criteria

* At the end all scripts have to be executable via build.py and build their respective packages.

## External dependencies

Extracted from build scripts in https://github.com/magma/magma/tree/master/third_party/build/bin.

- [x] Use build.py in the devcontainer to run the build scripts.
  - [x] Build `aioeventlet_build.sh`
  - [x] Build `aioh2-build.sh`
  - [x] Build `asn1c_build.sh`
  - [x] Build `bcc-build.sh`
  - [x] Build `folly_build.sh`
  - [x] Build `freediameter_build.sh`
  - [x] Build `gnutls_build.sh`
  - [x] Build `grpc_build.sh`
  - [x] Build `libfluid_build.sh`
  - [x] Build `liblfds_build.sh`
  - [x] Build `magma-cpp-redis_build.sh`
  - [x] Build `magma-libtacopie_build.sh`
  - [x] Build `nettle_build.sh`
  - [x] Build `ryu_build.sh`
  - [x] Build `prometheus-cpp_build.sh`
  - [x] Build `sentry-native-build.sh`
- [x] When the build is successful, test the installation (installation should be similar to how it's installed during a magma setup - also see existing CI job for nettle). 
  -> checked packages also installed in dev container via `sudo apt install ./*.deb`
- [x] Prepare internal review meeting.

### PR DESCRIPTION
**Pull request by mpfirrmann**
## Summary

This PR enables the build of Debian packages of Magma's third party dependencies. 
Closes #13466.

Note: The `aioh2` is not built with `fpm`, but the Debian file is created via its own `setup.py`. This leads to a different naming than the rest of the packages; `python3-aioh2_0.2.3-1_all.deb` vs. e.g. `oai-nettle_2.5-1ubuntu20.04_amd64.deb`.

## Test Plan

- All packages were built and installed in Magma's dev container.

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: third_party/build/build.py [hunk 1]
```diff
@@ -76,8 +77,8 @@ def strsplitbytes(the_bytes):
     return [item for item in re.split(r'\s+', decoded) if item]
 
 
-def buildscript(package_name):
-    return './bin/' + package_name + '_build.sh'
+def buildscript(package_name: str) -> str:
+    return glob.glob(f'./bin/{package_name}*.sh')[0]
```
Comment (jheidbrink) [inline third_party/build/build.py]: Is there a reason we want to support anything other than underscore? Could we just have the underscore as the only naming scheme?

File: third_party/build/bin/aioh2-build.sh [hunk 2]
```diff
@@ -11,37 +11,68 @@
...
+        python3-h2 \
+        python3-setuptools \
+        python3-distutils \
+        fakeroot
+
+}
+
+if_subcommand_exec
+
+sudo apt-get -y install \
```
Comment (jheidbrink) [inline third_party/build/bin/aioh2-build.sh]: I wonder why we need `apt-get install` here. Doesn't `build.py` parse `buildrequires` and install the required packages already?

**Review by jheidbrink — APPROVED** [review]

File: third_party/build/bin/bcc-build.sh [hunk 3]
```diff
@@ -22,16 +22,45 @@ WORK_DIR=/tmp/build-"$PKGNAME"
 VERSION="$PKGVERSION-$ITERATION"
 
 function buildrequires() {
```
Comment (nstng) [inline third_party/build/bin/bcc-build.sh:24]: Does this really work with `build.py` - my understanding is the file must be named `<dep>_build.sh` with an underscore. Same for `aioh2-build.sh`.

File: third_party/build/bin/bcc-build.sh [hunk 4]
```diff
@@ -22,16 +22,45 @@ WORK_DIR=/tmp/build-"$PKGNAME"
...
+      libluajit-5.1-dev
 }
 
-sudo apt install -y bison build-essential cmake flex git libedit-dev \
-  libllvm7 llvm-7-dev libclang-7-dev python zlib1g-dev libelf-dev libfl-dev python3-distutils
-sudo apt-get -y install luajit luajit-5.1-dev
-
 if_subcommand_exec
 
+sudo apt-get install -y \
```
Comment (nstng) [inline third_party/build/bin/bcc-build.sh]: is a manual install really necessary? Iiuc then `build.py` should handle this via `buildrequires()`.

File: third_party/build/build.py (hunk 1 above)
Comment (nstng) [reply third_party/build/build.py]: ah, I see - this explains my comments above. I would suggest to rename the files and keep the original naming scheme.

File: third_party/build/bin/gnutls_build.sh [hunk 5]
```diff
@@ -61,6 +66,32 @@ cd ${WORK_DIR}
 wget http://mirrors.dotsrc.org/gcrypt/gnutls/v3.1/gnutls-$PKGVERSION.tar.xz
 tar xf gnutls-$PKGVERSION.tar.xz
 cd gnutls-$PKGVERSION/
+
+patch gl/stdio-impl.h <<'EOF'
```
Comment (nstng) [inline third_party/build/bin/gnutls_build.sh:75]: What are the reasons for these patches? Also in `grpc_build.sh`

**Review by nstng — CHANGES_REQUESTED** [review]

File: third_party/build/build.py (hunk 1 above)
Comment (mpfirrmann) [reply third_party/build/build.py]: I don't have a strong opinion on this, tbh. Leaving the filenames as they are seemed the least intrusive way but renaming them is totally fine with me.

File: third_party/build/bin/bcc-build.sh (hunk 4 above)
Comment (mpfirrmann) [reply third_party/build/bin/bcc-build.sh]: Not necessary, removed.

File: third_party/build/build.py (hunk 1 above)
Comment (mpfirrmann) [reply third_party/build/build.py]: Renamed.

File: third_party/build/bin/aioh2_build.sh [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        python3-h2 \
+        python3-setuptools \
+        python3-distutils \
+        fakeroot
+
+}
+
+if_subcommand_exec
+
+sudo apt-get -y install \
```
Comment (jheidbrink) [inline third_party/build/bin/aioh2_build.sh]: We can also remove this `apt-get install` I think

File: third_party/build/bin/gnutls_build.sh (hunk 5 above)
Comment (mpfirrmann) [reply third_party/build/bin/gnutls_build.sh:75]: I added some information about the bugs which are fixed by these patches / their upstream fixes in newer versions.

File: third_party/build/bin/aioh2_build.sh (hunk 6 above)
Comment (mpfirrmann) [reply third_party/build/bin/aioh2_build.sh]: Removed, since this script was restructured a little bit.

**Review by nstng — APPROVED** [review]
lgtm

**Pull request merged**